src/main/java/org/java_websocket/extensions/permessage_deflate/PerMessageDeflateExtension.java

```diff
@@ -190,12 +214,16 @@ public void encodeFrame(Framedata inputFrame) {
       return;
     }
 
+    byte[] payloadData = inputFrame.getPayloadData().array();
+    if (payloadData.length < threshold) {
+      return;
+    }
     // Only the first frame's RSV1 must be set.
     if (!(inputFrame instanceof ContinuousFrame)) {
       ((DataFrame) inputFrame).setRSV1(true);
     }
 
-    deflater.setInput(inputFrame.getPayloadData().array());
+    deflater.setInput(payloadData);
```

Comment:
You can simplify this by using the ByteBuffer directly (setInput has an overload for it) and `remaining()` instead of `array().length` above